tools/oesign/main.c

```diff
@@ -589,27 +611,49 @@ int oesign(const char* enclave, const char* conffile, const char* keyfile)
         goto done;
     }
 
-    /* Load private key into memory */
-    if (_load_pem_file(keyfile, &pem_data, &pem_size) != 0)
+    if (engine_id)
     {
-        Err("Failed to load file: %s", keyfile);
-        goto done;
+        /* Initialize the SigStruct object */
+        if ((result = oe_sgx_sign_enclave_from_engine(
+                 &enc.hash,
+                 props.config.attributes,
+                 props.config.product_id,
+                 props.config.security_version,
+                 engine_id,
+                 engine_load_path,
+                 key_id,
+                 (sgx_sigstruct_t*)props.sigstruct)) != OE_OK)
+        {
+            Err("oe_sgx_sign_enclave_from_engine() failed: result=%s (%u)",
+                oe_result_str(result),
+                result);
+            goto done;
+        }
     }
-
-    /* Initialize the SigStruct object */
-    if ((result = oe_sgx_sign_enclave(
-             &enc.hash,
-             props.config.attributes,
-             props.config.product_id,
-             props.config.security_version,
-             pem_data,
-             pem_size,
-             (sgx_sigstruct_t*)props.sigstruct)) != OE_OK)
+    else
     {
-        Err("oe_sgx_sign_enclave() failed: result=%s (%u)",
-            oe_result_str(result),
-            result);
-        goto done;
+        /* Load private key into memory */
+        if (_load_pem_file(keyfile, &pem_data, &pem_size) != 0)
+        {
+            Err("Failed to load file: %s", keyfile);
+            goto done;
+        }
+
+        /* Initialize the SigStruct object */
```

Comment:
nit: I think it's actually "sigstruct" and not "SigStruct" isn't it?

Reply:
done